File: typescript-playwright/src/utils/matchers.ts
import { expect } from "../fixtures/fixtures";

export const PhotoMatcher = {
  base: expect.objectContaining({
    id: expect.any(Number),
    title: expect.any(String),
    author: expect.any(String),
    url: expect.any(String),
    fullUrl: expect.any(String),
    likes: expect.any(Number),
    views: expect.any(Number),
  }),
};

export const CategoryMatcher = {
  base: expect.objectContaining({
    id: expect.any(Number),
    name: expect.any(String),
  }),
};

export const SubcategoryMatcher = {
  base: expect.objectContaining({
    categoryId: expect.any(Number),
    id: expect.any(Number),
    name: expect.any(String),
  }),
};

File: python-pytest/src/utils/matchers.py
from __future__ import annotations

from dirty_equals import IsInt, IsPartialDict, IsStr

PhotoMatcher = {
    "base": IsPartialDict(
        id=IsInt,
        title=IsStr,
        author=IsStr,
        url=IsStr,
        fullUrl=IsStr,
        likes=IsInt,
        views=IsInt,
    ),
}

CategoryMatcher = {
    "base": IsPartialDict(
        id=IsInt,
        name=IsStr,
    ),
}

SubcategoryMatcher = {
    "base": IsPartialDict(
        categoryId=IsInt,
        id=IsInt,
        name=IsStr,
    ),
}
